src/ptf/dataplane.py

```diff
@@ -49,23 +49,38 @@
 #except:
 #    pass
 
+dump_pkts_on_mismatch = False
+
 def match_exp_pkt(exp_pkt, pkt):
     """
     Compare the string value of pkt with the string value of exp_pkt,
     and return True iff they are identical.  If the length of exp_pkt is
     less than the minimum Ethernet frame size (60 bytes), then padding
     bytes in pkt are ignored.
     """
+    is_match = False
     if isinstance(exp_pkt, mask.Mask):
         if not exp_pkt.is_valid():
+            logging.debug("match_exp_pkt: exp_pkt is not valid!")
             return False
-        return exp_pkt.pkt_match(pkt)
-    e = str(exp_pkt)
-    p = str(pkt)
-    if len(e) < 60:
-        p = p[:len(e)]
-    return e == p
-
+        is_match = exp_pkt.pkt_match(pkt)
+    else:
+        e = str(exp_pkt)
+        p = str(pkt)
+        if len(e) < 60:
+            p = p[:len(e)]
+        is_match = (e == p)
+
+    if (dump_pkts_on_mismatch and not is_match):
```

Comment:
There are multiple locations (at least 4, spread over in ptf-utils and ptf) where this function is called and there would be more usecases. Isn't it better to have a capability of dumping in this function?